x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/mapper/SemanticTextFieldMapper.java

```diff
@@ -89,11 +88,15 @@ protected Parameter<?>[] getParameters() {
 
         @Override
         public SemanticTextFieldMapper build(MapperBuilderContext context) {
-            return new SemanticTextFieldMapper(name(), new SemanticTextFieldType(name(), modelId.getValue(), meta.getValue()), copyTo);
+            return new SemanticTextFieldMapper(
+                name,
+                new SemanticTextFieldType(context.buildFullName(name), modelId.getValue(), meta.getValue()),
```

Comment:
This change fixes a bug where `semantic_text` fields that were not top-level fields (i.e. were part of an object field or declared as a multi-field) did not register with the proper fully-qualified field name.